mathesar_ui/src/stores/table-data/selection.ts

```diff
@@ -53,10 +62,13 @@ export class Selection {
 
   selectedCells: WritableSet<string>;
 
+  freezeSelection: boolean;
```

Comment:
Can you help me understand what `freezeSelection` is and why we need it?

Reply:
The idea here is to not let the user change the selection/make a new selection when this boolean is true. This is helpful when there is a long-running action on a particular selection and we don't want the user to update the selection until the action is finished.

Reply:
Cool. Makes sense. Thanks